src/lib/DropDownController.js

```diff
@@ -0,0 +1,54 @@
+class DropDownController {
+  constructor() {
+    if (DropDownController.instance instanceof DropDownController) {
+      return DropDownController.instance;
+    }
+
+    DropDownController.instance = this;
+  }
+
+  toggleDropDown(showDropDown) {
+    if (this.elem?.toggleDropDown) {
+      this.elem.toggleDropDown(showDropDown);
```

Comment:
Why are you still using `toggleDropDown`?

Correct me if I'm wrong, but it should be possible to achieve the desired behavior by passing the correct element to the `toggleDropDown` method. The rest of the styling can be taken care of by CSS.